test/routing/VersionRoutingRequestHandlerSpec.scala

```diff
@@ -0,0 +1,220 @@
+import com.typesafe.config.ConfigFactory
+import mocks.MockAppConfig
+import org.scalamock.handlers.CallHandler1
+import org.scalamock.scalatest.MockFactory
+import org.scalatest.{Inside, Matchers}
+import play.api.Configuration
+import play.api.http.HeaderNames.ACCEPT
+import play.api.http.{HttpConfiguration, HttpErrorHandler, HttpFilters}
+import play.api.libs.json.Json
+import play.api.mvc.{EssentialAction, _}
+import play.api.routing.Router
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import support.UnitSpec
+import v1.models.errors.{InvalidAcceptHeaderError, UnsupportedVersionError}
+
+class VersionRoutingRequestHandlerSpec extends UnitSpec with Matchers with MockFactory with Inside with MockAppConfig{
+  test =>
+
+  implicit private val actorSystem: ActorSystem = ActorSystem("test")
+  implicit private val mat: Materializer        = ActorMaterializer()
+
+  private val defaultRouter = mock[Router]
+  private val v1Router      = mock[Router]
+  private val v2Router      = mock[Router]
+  private val v3Router      = mock[Router]
+
+  private val routingMap = new VersionRoutingMap {
+    override val defaultRouter: Router = test.defaultRouter
+    override val map           = Map("1.0" -> v1Router, "2.0" -> v2Router, "3.0" -> v3Router)
+  }
+
+  class Test(implicit acceptHeader: Option[String]) {
+    val httpConfiguration = HttpConfiguration("context")
+    private val errorHandler      = mock[HttpErrorHandler]
+    private val filters           = mock[HttpFilters]
+    (filters.filters _).stubs().returns(Seq.empty)
+
+    MockedAppConfig.featureSwitch.returns(Some(Configuration(ConfigFactory.parseString("""
+                                                                           |version-1.enabled = true
+                                                                           |version-2.enabled = true
+                                                                         """.stripMargin))))
+
+    //noinspection ScalaDeprecation
+    val requestHandler: VersionRoutingRequestHandler =
+      new VersionRoutingRequestHandler(routingMap, errorHandler, httpConfiguration, mockAppConfig, filters, Action)
+
+    def stubHandling(router: Router, path: String)(handler: Option[Handler]): CallHandler1[RequestHeader, Option[Handler]] =
+      (router.handlerFor _)
+        .expects(where { r: RequestHeader =>
+          r.path == path
+        })
+        .returns(handler)
+
+    def buildRequest(path: String): RequestHeader =
+      acceptHeader
+        .foldLeft(FakeRequest("GET", path)) { (req, accept) =>
+          req.withHeaders((ACCEPT, accept))
+        }
+  }
+
+  "Routing requests with no version" should {
+    implicit val acceptHeader: None.type = None
+
+    handleWithDefaultRoutes(defaultRouter)
+  }
+
+  "Routing requests with valid version" should {
+    implicit val acceptHeader: Some[String] = Some("application/vnd.hmrc.1.0+json")
+
+    handleWithDefaultRoutes(defaultRouter)
```

Comment:
Should this test go to the non-default router?

Reply:
Yes..